var/spack/repos/builtin/packages/py-zfit-physics/package.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack.package import *
+
+
+class PyZfitPhysics(Package):
+    """Tools and models to extend zfit with physics specific content."""
+
+    homepage = "https://github.com/zfit/zfit-physics"
+    pypi = "zfit-physics/zfit_physics-0.7.0.tar.gz"
+
+    maintainers("jonas-eschle", "ikrommyd")
+    license("BSD-3-Clause", checked_by="jonas-eschle")
+
+    tags = ["likelihood", "statistics", "inference", "fitting", "hep"]
+
+    version("0.7.0", sha256="5d65becff7265a12d9b62a8476c5359e75ec10d6ac0fd84dfa39eb82b6693cda")
```

Comment:
I cannot confirm the version sha256 since the package's archive files in `https://files.pythonhosted.org/` do not match the pattern built-into the python package.

However, I *can* confirm the version sha256 (and that of the previous version)IF I remove the `pypi` property and replace it with `url = "https://pypi.org/simple/zfit-physics/zfit_physics-0.7.0.tar.gz"`.  

@adamjstewart Is swapping `pypi` for the simple `url` an acceptable workaround for python packages in this case?

Reply:
Hmm, that doesn't make sense. Can you investigate why they are different?

Reply:
The python package expects to derive the url from the `pypi` property (https://github.com/spack/spack/blob/44cb4eca933e53bd0af32e0715195c5c5797b689/lib/spack/spack/build_systems/python.py#L355).

But the pypi download url for `0.7.0` does not take that form (`https://files.pythonhosted.org/packages/22/76/6227f5fce65b9e4df6d9e773ccb670369f74e2b818224fe09b1c387fa897/zfit_physics-0.7.0.tar.gz`).

Reply:
The same is true for other versions of this and the package in #43697 .

Reply:
Perhaps the solution is to change the python package to use the "right" python url:

````
https://pypi.io/packages/source/{ package_name_first_letter }/{ package_name }/{ package_name }-{ package_version }.tar.gz
```

Reply:
I'm checking this idea out in https://github.com/spack/spack/pull/43754.

Reply:
Python package expects the url (https://github.com/spack/spack/blob/44cb4eca933e53bd0af32e0715195c5c5797b689/lib/spack/spack/build_systems/python.py#L355) to be `f"https://files.pythonhosted.org/packages/source/{cls.pypi[0]}/{cls.pypi}"`

Reply:
Well, it doesn't appear to help in this case but I need to revisit how python downloads files...

Reply:
Thanks for looking into it!

Reply:
The https://files.pythonhosted.org/packages/source URL should redirect to the https://files.pythonhosted.org/packages URL.

Reply:
It does through the browser but unfortunately it isn't working through Spack:  `==> Error: Could not find any remote versions for py-zfit-physics`.

Reply:
Ah, I finally figured out the bug.